pandas/tests/arrays/categorical/test_algos.py

```diff
@@ -94,14 +94,14 @@ def test_isin_empty(empty):
 
 def test_diff():
     s = pd.Series([1, 2, 3], dtype="category")
-    with tm.assert_produces_warning(FutureWarning):
+    with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
```

Comment:
You will have to restore those checks, they are important.

When a warning is raised, we want to raise them in the line that the user wrote, like the `Series` creation. This is what they are checking. If you run in your branch the code `s = pd.Series([1, 2, 3], dtype="category")` you'll see that this is not true anymore, and the warning will point to some internal code.

This is caused by using the doc decorator, and should be changes when we generate that `FutureWarning`. There we tell the stacklevel, the number of calls we want to go "up" to generate the warning there.

We're changing the doc decorator, so we'll have to wait until this can be merged, see: https://github.com/pandas-dev/pandas/pull/33160#pullrequestreview-400797773